composer/trainer/trainer.py

```diff
@@ -1107,10 +1089,17 @@ def _train_loop(self) -> None:
                 for scheduler in self.state.schedulers:
                     scheduler.step()
 
+            if self.train_metrics is not None:
```

Comment:
this was not in the original code, should it be introduced here?

Reply:
We do something similar for the evaluators, we we compute on each batch, and then afterwards, so I think it would make sense to do the same for the training dataloader. This also ensures that the metrics will get logged at least once per epoch if someone sets the LogLevel to EPOCH instead of batch.